scripts/lib/CIME/SystemTests/system_tests_common.py

```diff
@@ -350,7 +350,7 @@ def compare_env_run(self, expected=None):
         Compare env_run file to original and warn about differences
         """
         components = self._case.get_values("COMP_CLASSES")
-        f1obj = EnvRun(self._caseroot, "env_run.xml", components=components)
+        f1obj = self._case.get_env("run")
```

Comment:
Dangerous to create extra Env objects. That can gets us intro trouble with coherence issues.